tests/test_gates_gates.py

```diff
@@ -722,6 +725,12 @@ def test_cun(backend, name, params):
 
     backend.assert_allclose(final_state, target_state, atol=1e-6)
 
+    if name == "CRY":
+        matrix = Circuit(2)
+        matrix.add(decomposition)
```

Comment:
Maybe `decomposition` can be defined inside this if, or even do directly
```suggestion
        matrix.add(gate.decompose())
```
to avoid having another if above (line 701).

Reply:
I tried that but gate is being redefined as an ndarray before this line, this is why I added line 701.